### PR TITLE
Fix: various bugs

### PR DESCRIPTION
**Pull request by mykalmax**
1. Environments not getting updated after plan applied
2. Delay plan call after plan apply
3. Display nodes for CTEs when SQL has multiple `WITH`
Screenshot 2023-06-12 at 10 09 50 AM
4. Refetch environments, after plan promote step happened

### CODE REVIEW

File: web/server/console.py [hunk 1]
```diff
@@ -99,8 +99,7 @@ def log_test_results(
         self.queue.put_nowait(self._make_event(data, event="tests", ok=ok))
 
     def log_success(self, msg: str) -> None:
-        self.queue.put_nowait(self._make_event(msg))
-        self.stop_snapshot_progress()
+        self.queue.put_nowait(self._make_event(msg, event="promote-environment"))
```
Comment (mykalmax) [inline web/server/console.py:102]: `log_success` can be called not just when we `promoting`, console also does `log_success` when virtual env is executed. So maybe not the best name for this event, but calling it just `success` may be too generic

File: web/client/src/library/components/graph/help.ts [hunk 2]
```diff
@@ -452,18 +453,21 @@ function mergeConnections(
 
           connections.set(toNodeOrEdgeId(id, column), connectionTarget)
           connections.set(columnId, connectionSource)
+
+          addActiveEdges(
+            [
+              connectionTarget.left.map(id => toNodeOrEdgeId('right', id)),
+              connectionTarget.right.map(id => toNodeOrEdgeId('left', id)),
```
[the remark this answers is not in the document]
Comment (mykalmax) [reply web/client/src/library/components/graph/help.ts]: I've fixed some few things related to  your questions [5c0d85b](https://github.com/TobikoData/sqlmesh/pull/969/commits/5c0d85bbd3f4d67d9d662434686395ebcd38ee45)

File: web/client/src/library/pages/ide/RunPlan.tsx [hunk 3]
```diff
@@ -438,7 +398,13 @@ function SelectEnvironemnt({
             disabled={disabled}
             className={clsx(className)}
           >
-            <span className="block overflow-hidden truncate">
+            <span
+              className={clsx(
+                'block overflow-hidden truncate',
+                (environment.isLocal || disabled) && 'text-neutral-500',
+                environment.isSyncronized && 'text-primary-500',
```
Comment (vchan) [inline web/client/src/library/pages/ide/RunPlan.tsx:405]: I just noticed Synchronized is misspelled.

File: web/client/src/library/components/graph/Graph.tsx [hunk 4]
```diff
@@ -944,8 +954,8 @@ function ModelNode({
             showColumns ? 'rounded-t-lg' : 'rounded-lg',
             isCTE ? 'bg-accent-500' : 'bg-secondary-100 dark:bg-primary-900',
           )}
-          hasLeft={sourcePosition === Position.Left}
-          hasRight={targetPosition === Position.Right}
+          hasLeft={sourcePosition === Position.Right}
+          hasRight={targetPosition === Position.Left}
```
Comment (vchan) [inline web/client/src/library/components/graph/Graph.tsx:958]: Confirming that left and right is correctly assigned here.

**Review by vchan — APPROVED** [review]

**Pull request merged**